From fc3d1b2f61a4cba66ba8f952ad332b83e9baf308 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Sun, 17 Mar 2024 20:32:57 -0700
Subject: [PATCH 06/10] i40e: drop the weight argument from netif_napi_add

This patch accommodates commit b48b89f9c189 ("net: drop the weight argument
from netif_napi_add") merged in the v6.1 development cycle, by applying a
similar change to the out-of-tree i40e driver to resolve the compilation
errors encountered with the v6.6 kernel, which are listed below.

References:
* https://git.yoctoproject.org/linux-yocto/commit/?h=b48b89f9c189

Resolved compilation errors:

src/i40e_main.c:12759:3: error: too many arguments to function netif_napi_add
12759 |   netif_napi_add(vsi->netdev, &q_vector->napi,
      |   ^~~~~~~~~~~~~~
In file included from /usr/src/linux-headers-6.6.0-1-common/include/net/sock.h:46,
                 from /usr/src/linux-headers-6.6.0-1-common/include/linux/tcp.h:19,
                 from /usr/src/linux-headers-6.6.0-1-common/include/net/tcp.h:20,
                 from /<<PKGBUILDDIR>>/src/i40e.h:7,
                 from /<<PKGBUILDDIR>>/src/i40e_main.c:8:
/usr/src/linux-headers-6.6.0-1-common/include/linux/netdevice.h:2642:1: note: \
 declared here
 2642 | netif_napi_add(struct net_device *dev, struct napi_struct *napi,

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 src/i40e_main.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/i40e_main.c b/src/i40e_main.c
index 1319cfa..4c64e3e 100644
--- a/src/i40e_main.c
+++ b/src/i40e_main.c
@@ -12756,8 +12756,7 @@ static int i40e_vsi_alloc_q_vector(struct i40e_vsi *vsi, int v_idx)
 	cpumask_copy(&q_vector->affinity_mask, cpu_possible_mask);
 #endif
 	if (vsi->netdev)
-		netif_napi_add(vsi->netdev, &q_vector->napi,
-			       i40e_napi_poll, NAPI_POLL_WEIGHT);
+		netif_napi_add(vsi->netdev, &q_vector->napi, i40e_napi_poll);
 
 	/* tie q_vector and vsi together */
 	vsi->q_vectors[v_idx] = q_vector;
-- 
2.42.0

